File: question/neetcode150/array_and_hashing/longest-consecutive-sequence.py
from typing import List

class Solution:
    def longestConsecutive(self, a: List[int]) -> int:
        s = set(a)

        ans = 0
        for n in s:
            if n-1 in s:
                continue
            
            i = 0
            while n+i in s:
                ans = max(ans, i+1)
                i+=1

        return ans

File: question/neetcode/array_and_hashing/longest-consecutive-sequence.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s;

        for(auto x: nums) s.insert(x);

        int a = 0;
        for(auto x: s){
            if(s.count(x-1)) continue;
            int c = 0;

            int j=x;
            while(s.count(j)){
                c++;
                j++;
            }

            a = max(a,c);
        }

        return a;
    }
};